### PR TITLE
Remove pytket-quantinuum dependency.

### PR DESCRIPTION
**Pull request by cqc-alec**
And remove the `tket_optimization_level` argument to `pytket_to_phir()`. Add a `gateset` field to the `Machine` class and populate the `QtmMachine` instances with the current supported gatesets. Apply a standard rebase pass after decomposing boxes (i.e. optimization level 0).

Create a dataclass for machine timing data to get around ruff's (quite reasonable) complaint that there were too many arguments to `Machine.__init__()`.

### CODE REVIEW

File: tests/test_placement.py
Comment (qartik) [inline tests/test_placement.py:24]: I wonder if we can create a constant for this gate set with a descriptive name, perhaps even to be used as the default in the `Machine` class?

File: pytket/phir/qtm_machine.py
Comment (qartik) [inline pytket/phir/qtm_machine.py:34]: Since this is a common gateset, perhaps this should also be exposed as a constant from this module or the `QtmMachine` class.

File: pytket/phir/cli.py
Comment (qartik) [inline pytket/phir/cli.py:55]: As this argument is removed, could you update the readme with the output of `phirc -h`? You may need to issue `make dev-all` before being able to use the CLI.

**Review by qartik — CHANGES_REQUESTED** [review]
Thanks, this is looking great. Great idea to abstract timings into its own data class.

README update is the only needed change, rest are suggestions.

File: tests/test_placement.py
Comment (cqc-alec) [reply tests/test_placement.py:24]: This was an arbitrary choice -- other universal gatesets would have done as well -- but good idea to make it a constant to avoid repetition. I'm not sure it's worth making it a default for `Machine` -- though I agree it makes sense to make the quantinuum gateset the default for `QtmMachine`.

Comment (qartik) [reply tests/test_placement.py:24]: > This was an arbitrary choice -- other universal gatesets would have done as well

Ah, I misunderstood it as being critical. In that case, I agree it's not required to be the default.

File: tests/e2e_test.py
Comment (qartik) [inline tests/e2e_test.py:22]: Does it make sense to add `tk_gateset` as a constant under the `Machine` module?

This test is somewhat of dead code as it's currently not exercised with `make tests`, so I don't really care as much. However, I was thinking of adding `python tests/e2e_test.py` under the `make tests` command.

**Review by qartik — DISMISSED** [review]
LGTM. I'd wait for @neal-erickson's review before merging this. Also, left a minor comment.

File: pytket/phir/rebasing/rebaser.py
Comment (neal-erickson) [inline pytket/phir/rebasing/rebaser.py:24]: This is roughly equivalent to the Quantinuum gateset, do we want to rebase to that in the case of no machine - seems like it might reduce the flexibility of allowing this to be run without any Quantinuum specificity.

**Review by neal-erickson — DISMISSED** [review]
One question about approach to the "no machine" handling but otherwise looks good to me.

File: tests/e2e_test.py
Comment (cqc-alec) [reply tests/e2e_test.py]: I changed this to use `QTM_DEFAULT_GATESET` instead.

File: pytket/phir/rebasing/rebaser.py
Comment (cqc-alec) [reply pytket/phir/rebasing/rebaser.py]: I changed the code to use `QTM_DEFAULT_GATESET` explicitly. I don't think this reduces flexibility, as one can specify a machine with another gateset if desired.

**Review by qartik — APPROVED** [review]

**Pull request merged**
